src/util/auth.rs

```diff
@@ -1,36 +1,58 @@
-use actix_multipart::Multipart;
-use actix_web::dev::ServiceRequest;
-use actix_web::web::Bytes;
-use actix_web::{error, Error};
-use actix_web_httpauth::extractors::basic::BasicAuth;
-use futures::TryStreamExt;
-
 use crate::args::ARGS;
+use crate::error_handling::AppError;
+use axum::extract::{Multipart, Request};
+use axum::middleware::Next;
+use axum::response::Response;
+use base64::engine::general_purpose;
+use base64::Engine;
 
-pub async fn auth_validator(
-    req: ServiceRequest,
-    creds: BasicAuth,
-) -> Result<ServiceRequest, (Error, ServiceRequest)> {
-    match (
-        ARGS.auth_basic_username.as_ref(),
-        ARGS.auth_basic_password.as_ref(),
-        creds.password(),
-    ) {
-        (Some(conf_user), Some(conf_pwd), Some(cred_pwd))
-            if creds.user_id() == conf_user && conf_pwd == cred_pwd =>
-        {
-            Ok(req)
-        }
-        _ => Err((error::ErrorBadRequest("Invalid login details."), req)),
+pub async fn auth_validator(req: Request, next: Next) -> Response {
+    let (username, password) = req
+        .headers()
+        .get("authorization")
+        .and_then(|auth| auth.to_str().ok())
+        .and_then(|auth_str| {
+            let parts: Vec<&str> = auth_str.splitn(2, ' ').collect();
```

Comment:
looks like a lot of work! Lets keep it but make an issue to in the future move to https://crates.io/crates/axum-login.